### ISSUE TITLE
Use pypsa's optimize_with_rolling_horizon() functionality

### ISSUE THREAD

**Issue by ulfmueller** [openego/eTraGo#664]
in particular for our market simulation...
see [pypsa](https://github.com/PyPSA/PyPSA/blob/b066cb5d7af7b30861a2b76e486ec3910778e9c9/pypsa/optimization/abstract.py#L254)
depends on #663

**Comment by ulfmueller** [comment]
First running example is implemented [here](https://github.com/openego/eTraGo/blob/85e8a9383dd657aaaad402b76bf13b62217dc1d0/etrago/execute/market_optimization.py#L73-L74). 
But details remain to be done:

- [ ] define a suitable planning horizon and overlap primarily with a purpose of realistic assumption (and not with a purpose of reducing computational burden). Try to get NEP insights and/or do more literature research, consult market experts.

**Comment by ClaraBuettner** [comment]
When I understand the code correctly, there is currently no extra_functionality set in the rolling_horizont optimization. WE would need that at least for defining the CHP heat and electricity dispatch. 
We should find out if it is possible to add extra_functionalities to the rolling_horizont optimization and if yes, we need to use the constraints for linopy. They are already translated, but these changes are not on dev yet.

**Comment by ClaraBuettner** [comment]
> When I understand the code correctly, there is currently no extra_functionality set in the rolling_horizont optimization. We would need that at least for defining the CHP heat and electricity dispatch. We should find out if it is possible to add extra_functionalities to the rolling_horizont optimization and if yes, we need to use the constraints for linopy. They are already translated, but these changes are not on dev yet.

Update: We can add extra_functionalities to the rolling_horizont optimization easily. So we just need the translated constraints from the other branch. I will do this soon.

**Comment by ClaraBuettner** [comment]
The comments listed above are already included in eTraGo. There might be some updates in the future, but I think it is better to open a new issue by then. So I will close this issue now.
